src/accordion/panel.js

```diff
@@ -90,62 +113,32 @@ class Panel extends React.Component<PanelPropsT, {isFocusVisible: boolean}> {
     } = this.props;
 
     const sharedProps = this.getSharedProps();
-    const {
-      PanelContainer: PanelContainerOverride,
-      Header: HeaderOverride,
-      Content: ContentOverride,
-      ToggleIcon: ToggleIconOverride,
-    } = overrides;
-
-    const isIconOverriden =
-      ToggleIconOverride &&
-      (typeof ToggleIconOverride === 'function' ||
-        typeof ToggleIconOverride.component === 'function');
-
-    const [PanelContainer, panelContainerProps] = getOverrides(
-      PanelContainerOverride,
-      StyledPanelContainer,
-    );
-    const [Header, headerProps] = getOverrides(HeaderOverride, StyledHeader);
-    const [Content, contentProps] = getOverrides(
-      ContentOverride,
-      StyledContent,
-    );
-    const [ToggleIcon, toggleIconProps] = getOverrides(
-      ToggleIconOverride,
-      StyledToggleIcon,
-    );
-
-    const toggleIconOverrides = mergeOverrides(
-      {Svg: ToggleIcon},
-      // $FlowFixMe
-      {Svg: ToggleIconOverride},
-    );
+    const ToggleIconComponent = expanded ? CheckIndeterminateIcon : PlusIcon;
 
-    // it's a bit tricky ¯\_(ツ)_/¯
-    // we only want to use the theme overrides, if it was not override locally
-    const ToggleIconComponent = isIconOverriden
-      ? ToggleIcon
-      : expanded
-      ? CheckIndeterminateIcon
-      : PlusIcon;
+    const HeaderOverrides = mergeOverrideResources(overrides.Header || {}, {
+      props: {
+        // constrain focus handlers, overrides handled in component methods
+        onFocus: this.handleFocus,
+        onBlur: this.handleBlur,
+      },
+    });
 
     return (
       <LocaleContext.Consumer>
         {locale => (
-          <PanelContainer {...sharedProps} {...panelContainerProps}>
+          <PanelContainer {...sharedProps} override={overrides.PanelContainer}>
             <Header
               tabIndex={0}
               role="button"
               aria-expanded={expanded}
               aria-disabled={disabled || null}
               {...sharedProps}
-              {...headerProps}
               {...(ariaControls ? {'aria-controls': ariaControls} : {})}
               onClick={this.onClick}
               onKeyDown={this.onKeyDown}
-              onFocus={forkFocus(headerProps, this.handleFocus)}
-              onBlur={forkBlur(headerProps, this.handleBlur)}
+              onFocus={this.handleFocus}
+              onBlur={this.handleBlur}
```

Comment:
Since the above props are merged in the `HeaderOverrides` do we still need it here?

Reply:
This is so that if `override.props` is a function, consuming applications could read the initial value